source/Coop.IntegrationTests/MobileParties/TroopSync.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Coop.IntegrationTests.Environment;
+using Coop.IntegrationTests.Environment.Instance;
+using GameInterface.Services.MobileParties.Messages;
+
+namespace Coop.IntegrationTests.MobileParties
+{
+    public class TroopSync
+    {
+        internal TestEnvironment TestEnvironment { get; }
+
+        public TroopSync()
+        {
+            // Creates a test environment with 1 server and 2 clients by default
+            TestEnvironment = new TestEnvironment();
+        }
+
+        /// <summary>
+        /// Verify sending TroopIndexAdded on one client
+        /// Triggers AddTroopIndex on all other clients
+        /// </summary>
+        [Fact]
+        public void TroopIndexAdded_Publishes_AllClients()
```

Comment:
I'm not sure how AI is controlled with this, is it not?

Reply:
From what I have seen, it does